tests/generator/BGenTool.cs

```diff
@@ -71,6 +72,11 @@ public void AddTestApiDefinition (string filename)
 			ApiDefinitions.Add (Path.Combine (Configuration.SourceRoot, "tests", "generator", filename));
 		}
 
+		public void AddExtraSources (string filename)
```

Comment:
The name is somewhat confusing since what the function is named is not what it does. Initially from the name I was confused as to why we needed this since the list was public. Then looking at the source, it was clear that it was really AddToExtraSourcesRelativeToGenerator.

Since I see the same code in `AddTestApiDefinition`, maybe it would be better to have something like this:
```
string PathRelativeToGenerator(string filename) => Path.Combine(Configuration.SourceRoot, "tests", "generator", filename); // should "tests" and "generator" be constants?
```
And then just use `ExtraSources.Add()` since it's a public field, or use the relative code in the convenience Add. Either way, I'm not happy with the name.

Reply:
@stephen-hawley I made some changes, please have another look